tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,705 @@
+        let pubkeyvalid2 = "valid2";
+        let testnameone = "one";
+        let testnametwo = "two";
+        let assert_keys = ["valid1"].to_vec();
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        let assert_keys = ["valid1", "valid2"].to_vec();
+
+        // "update-ssh-keys --add two valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnametwo,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+    }
+
+    #[test]
+    fn test_del_one() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid1 = "valid1";
+        let testnameone = "one";
+        let assert_keys = ["valid1"].to_vec();
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
+
+        // "update-ssh-keys --delete one valid1.pub"
+        match del_one_ssh_key(ssh_dir, pubkeyvalid1, testnameone) {
+            Ok(_) => panic!("unexpected test success"),
+            Err(err) => {
+                println!("update_ssh_keys --delete failed");
+
+                assert!(format!("{}", err).contains("no keys found"));
+
+                // NOTE: it is not possible to check for fingerprint, as key is not available in
+                // the context.
+                //                assert!(key.to_fingerprint_string().contains(&*FINGERPRINTS[pubkeyvalid1]));
+
+                let authkeyone: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/{}", ssh_dir, testnameone));
+
+                assert!(!authkeyone.exists());
+
+                let mut svec: Vec<&str> = Vec::new();
+                svec.push(pubkeyvalid1);
+                assert_has_keys(ssh_dir, svec, true);
+            }
+        }
+    }
+
+    #[test]
+    fn test_del_two() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
```

Comment:
Same here, `run_test_add_two(&ctx)` instead of the duplication

Reply:
Called `run_test_add_two(&ctx.ssh_dir)`.